client/castApp/src/castinit.c

```diff
@@ -95,91 +94,54 @@ void addReccasterEnvVars(caster_t* self, int argc, char **argv)
         epicsMutexUnlock(self->lock);
         return;
     }
-    int new_extra_envs_size = self->num_extra_envs + argc;
-    int num_new_extra_envs = self->num_extra_envs;
 
-    char **new_extra_envs = calloc(new_extra_envs_size, sizeof(*new_extra_envs));
-    if(new_extra_envs == NULL) {
-        errlogSevPrintf(errlogMajor, "Error in memory allocation of new_extra_envs from addReccasterEnvVars\n");
-        epicsMutexUnlock(self->lock);
-        return;
-    }
-    /* copy self->extra_envs into new_extra_envs with room for new envs */
-    for(i=0; i < self->num_extra_envs; i++) {
-        if((new_extra_envs[i] = strdup(self->extra_envs[i])) == NULL) {
-            errlogSevPrintf(errlogMinor, "strdup error for copying %s to new_extra_envs[%zu] from addReccasterEnvVars\n", self->extra_envs[i], i);
-            ret = 1;
-            break;
-        }
-    }
     int found_dup;
     /* sanitize input - check for dups and empty args */
-    if(!ret) {
-        for(i=0; i < argc; i++) {
-            if(argv[i] == NULL) {
-                errlogSevPrintf(errlogMinor, "Arg is NULL for addReccasterEnvVars\n");
-                continue;
-            }
-            else if(argv[i][0] == '\0') {
-                errlogSevPrintf(errlogMinor, "Arg is empty for addReccasterEnvVars\n");
-                continue;
-            }
-            found_dup = 0;
-            /* check if dup in self->default_envs */
-            for(j = 0; default_envs[j]; j++) {
-                if(strcmp(argv[i], default_envs[j]) == 0) {
-                    found_dup = 1;
-                    errlogSevPrintf(errlogMinor, "Env var %s is already in env list sent by reccaster by default\n", argv[i]);
-                    break;
-                }
-            }
-            if(found_dup) {
-                continue;
-            }
-            /* check if dup in self->extra_envs */
-            for(j = 0; j < num_new_extra_envs; j++) {
-                if(new_extra_envs[j] == NULL) {
-                    continue;
-                }
-                if(strcmp(argv[i], new_extra_envs[j]) == 0) {
-                    found_dup = 1;
-                    errlogSevPrintf(errlogMinor, "Env var %s is already in extra_envs list\n", argv[i]);
-                    break;
-                }
-            }
-            if(found_dup) {
-                continue;
+    for(i=0; i < argc; i++) {
```

Comment:
Lots of duplication in this loop and in the one for exclude patterns, just a thought...